vmdb/app/models/miq_worker.rb

```diff
@@ -88,7 +88,7 @@ def self.server_scope(server_id = nil)
       server = MiqServer.my_server
       server_id = server.id unless server.nil?
     end
-    where(:miq_server_id => server_id)
+    where(:miq_server_id => server_id, :type => self.to_s)
```

Comment:
It's weird that server_scope would do this directly.  I would think server_scope would just do the server piece.  The caller should tack on the `.where(:type => self.to_s)` or we should add another scope that does it.